src/main/java/org/mockito/internal/verification/MockAwareVerificationMode.java

```diff
@@ -4,21 +4,46 @@
  */
 package org.mockito.internal.verification;
 
+import java.util.Set;
+
 import org.mockito.internal.verification.api.VerificationData;
+
+import org.mockito.listeners.VerificationListener;
 import org.mockito.verification.VerificationMode;
+import org.mockito.verification.VerificationSucceededEvent;
 
 public class MockAwareVerificationMode implements VerificationMode {
 
     private final Object mock;
     private final VerificationMode mode;
+    private final Set<VerificationListener> listeners;
 
-    public MockAwareVerificationMode(Object mock, VerificationMode mode) {
+    public MockAwareVerificationMode(Object mock, VerificationMode mode, Set<VerificationListener> listeners) {
         this.mock = mock;
         this.mode = mode;
+        this.listeners = listeners;
     }
 
     public void verify(VerificationData data) {
-        mode.verify(data);
+        try {
+            mode.verify(data);
+
+            for (VerificationListener listener : listeners) {
+                listener.onVerificationSucceeded(new VerificationSucceededEvent(mock, mode, data));
+            }
+        } catch (RuntimeException e) {
+            notifyExceptionListeners(e);
+            throw e;
+        } catch (Error e) {
```

Comment:
Do you have unit tests for both cases, errors and exception emitted during verification? If no, please add, if yes, ignore my request :)